infra/cifuzz/fuzz_target.py

```diff
@@ -134,3 +179,65 @@ def get_test_case(self, error_string):
     if match:
       return os.path.join(self.out_dir, match.group(1))
     return None
+
+  def get_lastest_build_version(self):
+    """Gets the latest OSS-Fuzz build version for a projects fuzzers.
+
+    Returns:
+      A string with the latest build version or None.
+    """
+    if not self.project_name:
+      return None
+    sanitizer = 'address'
+    version = '{project_name}-{sanitizer}-latest.version'.format(
+        project_name=self.project_name, sanitizer=sanitizer)
+    version_url = url_join(GCS_BASE_URL, self.project_name, version)
+    try:
+      response = urllib.request.urlopen(version_url)
+    except urllib.error.HTTPError:
+      logging.error('Error getting the lastest build version for %s.',
+                    self.project_name)
+      return None
+    return response.read().decode('UTF-8')
+
+  def download_oss_fuzz_build(self):
+    """Downloads the latest OSS-Fuzz build from GCS.
+
+    Returns:
+      A path to where the OSS-Fuzz build is located, or None.
+    """
+    if not os.path.exists(self.out_dir):
+      logging.error('Out directory %s does not exist.', self.out_dir)
+      return None
+    if not self.project_name:
+      return None
+    build_dir = os.path.join(self.out_dir, 'oss_fuzz_latest', self.project_name)
+    if os.path.exists(os.path.join(build_dir, self.target_name)):
+      return build_dir
+    os.makedirs(build_dir, exist_ok=True)
+    latest_build_str = self.get_lastest_build_version()
+    if not latest_build_str:
+      return None
+
+    oss_fuzz_build_url = url_join(GCS_BASE_URL, self.project_name,
+                                  latest_build_str)
+    try:
+      response = urllib.request.urlopen(oss_fuzz_build_url)
+    except urllib.error.HTTPError:
+      logging.error('Unable to download build from: %s.', oss_fuzz_build_url)
+      return None
+    with zipfile.ZipFile(io.BytesIO(response.read())) as zip_file:
```

Comment:
We shouldn't read the entire build into memory. We have some pretty big builds, so it's better to write the zip to somewhere on disk instead.

Reply:
Fixed.